### PR TITLE
Renaming RuntimeEventSource to NativeRuntimeEventSource

### PR DESCRIPTION
**Pull request by sywhang**
This is part of #20372.

The new EventSource that contains performance counters should be named "RuntimeEventSource" instead of "ManagedRuntimeEventSource", but we are already using RuntimeEventSource to emit events from the native runtime. I'm renaming the existing ```RuntimeEventSource``` to ```NativeRuntimeEventSource``` so that the existing name can be used for the new EventSource for performance counters.

### CODE REVIEW

**Comment by vancem** [comment]
This change is all about making sure you didn't miss anything.    If you have not already do a global search for RuntimeEventSource in all files ("findstr /s /p RuntimeEventSource  * ").  

If that comes up clean and tests work then I am OK with it.  

@brianrob

File: src/System.Private.CoreLib/src/System/Diagnostics/Eventing/DotNETRuntimeEventSource.cs [hunk 1]
```diff
@@ -6,19 +6,19 @@ namespace System.Diagnostics.Tracing
 {
 #if FEATURE_PERFTRACING
     /// <summary>
-    /// RuntimeEventSource is an EventSource that represents the ETW/EventPipe events emitted by the native runtime.
-    /// Most of RuntimeEventSource is auto-generated by scripts/genRuntimeEventSources.py based on the contents of the Microsoft-Windows-DotNETRuntime provider.
+    /// NativeRuntimeEventSource is an EventSource that represents the ETW/EventPipe events emitted by the native runtime.
+    /// Most of NativeRuntimeEventSource is auto-generated by scripts/genRuntimeEventSources.py based on the contents of the Microsoft-Windows-DotNETRuntime provider.
```
Comment (jkotas) [inline src/System.Private.CoreLib/src/System/Diagnostics/Eventing/DotNETRuntimeEventSource.cs]: Could you please rename the file to `NativeRuntimeEventSource.cs`. Our convention is that the type name should match the file name.

File: src/System.Private.CoreLib/src/System/Diagnostics/Eventing/DotNETRuntimeEventSource.cs [hunk 2]
```diff
@@ -6,19 +6,19 @@ namespace System.Diagnostics.Tracing
 {
 #if FEATURE_PERFTRACING
```
Comment (jkotas) [inline src/System.Private.CoreLib/src/System/Diagnostics/Eventing/DotNETRuntimeEventSource.cs]: Could you please add `Condition` to the .csproj file, and remove this end-to-end ifdef.

**Comment by jorive** [comment]
There are some comments in CoreFx about this class (RuntimeEventSource) that should probably be renamed too.

File: src/System.Private.CoreLib/System.Private.CoreLib.csproj [hunk 3]
```diff
@@ -305,7 +305,7 @@
   </ItemGroup>
   <ItemGroup>
     <Compile Include="$(BclSourcesRoot)\System\Diagnostics\Eventing\XplatEventLogger.cs" Condition="'$(FeatureXplatEventSource)' == 'true'" />
-    <Compile Include="$(IntermediateOutputPath)..\eventing\DotNETRuntimeEventSource.cs" />
+    <Compile Include="$(IntermediateOutputPath)..\eventing\NativeRuntimeEventSource.cs" Condition="'$(FeaturePerfTracing)' == 'true'"/>
```
Comment (jkotas) [inline src/System.Private.CoreLib/System.Private.CoreLib.csproj]:
```suggestion
    <Compile Include="$(IntermediateOutputPath)..\Eventing\NativeRuntimeEventSource.cs" Condition="'$(FeaturePerfTracing)' == 'true'"/>
```

**Pull request merged**